Building with Xcode 15's cmdline tools does not work (watchOS 10 SDK)

Will try to see what I can do. If I find a fix, I will open a pull request. macOS builds fine, did not try iOS/tvOS.

Xcode 15.0 (15A240d) (RC)
Using latest cmake 3.27/3.27.4, etc required tools written in README. OpenSSL builds fine (all platforms)

Xcode 14.3 / watchOS 9.4 SDK builds just fine.
```
-- Installing: /Users/vincent/Developer/Resources/Pigeon/Dependencies/TDLib/1.8.18-e79f540-2/example/ios/build/install-macOS/lib/libtddb.a
Platform = watchOS
iOS platform = WATCHOS
OpenSSL path = /Users/vincent/Developer/Resources/Pigeon/Dependencies/TDLib/1.8.18-e79f540-2/example/ios/third_party/openssl/watchOS
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Toolchain using default iOS SDK: /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk
-- The iOS architectures: armv7k;arm64_32;arm64
-- The iOS architectures: armv7k;arm64_32;arm64
-- The CXX compiler identification is AppleClang 15.0.0.15000040
-- The C compiler identification is AppleClang 15.0.0.15000040
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - failed
-- Check for working CXX compiler: /usr/bin/g++
-- Check for working CXX compiler: /usr/bin/g++ - works
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: /usr/bin/gcc
-- Check for working C compiler: /usr/bin/gcc - works
-- Detecting C compile features
-- Detecting C compile features - done
-- Could NOT find ccache (this is NOT an error)
-- Found OpenSSL: /Users/vincent/Developer/Resources/Pigeon/Dependencies/TDLib/1.8.18-e79f540-2/example/ios/third_party/openssl/watchOS/include /Users/vincent/Developer/Resources/Pigeon/Dependencies/TDLib/1.8.18-e79f540-2/example/ios/third_party/openssl/watchOS/lib/libcrypto.a;/Users/vincent/Developer/Resources/Pigeon/Dependencies/TDLib/1.8.18-e79f540-2/example/ios/third_party/openssl/watchOS/lib/libssl.a
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Performing Test HAVE_STD14
-- Performing Test HAVE_STD14 - Success
-- Performing Test HAVE_CXX_FLAG_WALL
-- Performing Test HAVE_CXX_FLAG_WALL - Success
-- Performing Test HAVE_CXX_FLAG_WEXTRA
-- Performing Test HAVE_CXX_FLAG_WEXTRA - Success
-- Performing Test HAVE_CXX_FLAG_WIMPLICIT_FALLTHROUGH_2
-- Performing Test HAVE_CXX_FLAG_WIMPLICIT_FALLTHROUGH_2 - Failed
-- Performing Test HAVE_CXX_FLAG_WPOINTER_ARITH
-- Performing Test HAVE_CXX_FLAG_WPOINTER_ARITH - Success
-- Performing Test HAVE_CXX_FLAG_WCAST_QUAL
-- Performing Test HAVE_CXX_FLAG_WCAST_QUAL - Success
-- Performing Test HAVE_CXX_FLAG_WSIGN_COMPARE
-- Performing Test HAVE_CXX_FLAG_WSIGN_COMPARE - Success
-- Performing Test HAVE_CXX_FLAG_WDUPLICATED_BRANCHES
-- Performing Test HAVE_CXX_FLAG_WDUPLICATED_BRANCHES - Failed
-- Performing Test HAVE_CXX_FLAG_WDUPLICATED_COND
-- Performing Test HAVE_CXX_FLAG_WDUPLICATED_COND - Failed
-- Performing Test HAVE_CXX_FLAG_WALLOC_ZERO
-- Performing Test HAVE_CXX_FLAG_WALLOC_ZERO - Failed
-- Performing Test HAVE_CXX_FLAG_WLOGICAL_OP
-- Performing Test HAVE_CXX_FLAG_WLOGICAL_OP - Failed
-- Performing Test HAVE_CXX_FLAG_WTAUTOLOGICAL_COMPARE
-- Performing Test HAVE_CXX_FLAG_WTAUTOLOGICAL_COMPARE - Success
-- Performing Test HAVE_CXX_FLAG_WVLA
-- Performing Test HAVE_CXX_FLAG_WVLA - Success
-- Performing Test HAVE_CXX_FLAG_WNON_VIRTUAL_DTOR
-- Performing Test HAVE_CXX_FLAG_WNON_VIRTUAL_DTOR - Success
-- Performing Test HAVE_CXX_FLAG_WUNUSED_PARAMETER
-- Performing Test HAVE_CXX_FLAG_WUNUSED_PARAMETER - Success
-- Performing Test HAVE_CXX_FLAG_WCONVERSION
-- Performing Test HAVE_CXX_FLAG_WCONVERSION - Success
-- Performing Test HAVE_CXX_FLAG_WSIGN_CONVERSION
-- Performing Test HAVE_CXX_FLAG_WSIGN_CONVERSION - Success
-- Performing Test HAVE_CXX_FLAG_WCXX14_COMPAT_PEDANTIC
-- Performing Test HAVE_CXX_FLAG_WCXX14_COMPAT_PEDANTIC - Success
-- Performing Test HAVE_CXX_FLAG_WDEPRECATED
-- Performing Test HAVE_CXX_FLAG_WDEPRECATED - Success
-- Performing Test HAVE_CXX_FLAG_WUNUSED_COMMAND_LINE_ARGUMENT
-- Performing Test HAVE_CXX_FLAG_WUNUSED_COMMAND_LINE_ARGUMENT - Success
-- Performing Test HAVE_CXX_FLAG_QUNUSED_ARGUMENTS
-- Performing Test HAVE_CXX_FLAG_QUNUSED_ARGUMENTS - Success
-- Performing Test HAVE_CXX_FLAG_WUNKNOWN_WARNING_OPTION
-- Performing Test HAVE_CXX_FLAG_WUNKNOWN_WARNING_OPTION - Success
-- Performing Test HAVE_CXX_FLAG_WODR
-- Performing Test HAVE_CXX_FLAG_WODR - Success
-- Performing Test HAVE_CXX_FLAG_FLTO_ODR_TYPE_MERGING
-- Performing Test HAVE_CXX_FLAG_FLTO_ODR_TYPE_MERGING - Failed
-- Could NOT find Git (missing: GIT_EXECUTABLE) 
-- Git state: e79f5409378e3e4b56d870619f3154ba2842a996
-- Found ZLIB: /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/lib/libz.tbd (found version "1.2.12")  
-- Found ZLIB: /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/lib/libz.tbd
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- Configuring done (8.0s)
-- Generating done (0.1s)
-- Build files have been written to: /Users/vincent/Developer/Resources/Pigeon/Dependencies/TDLib/1.8.18-e79f540-2/example/ios/build/build-watchOS
[  0%] Building CXX object tdtl/CMakeFiles/tdtl.dir/td/tl/tl_config.cpp.o
[  0%] Building CXX object tdtl/CMakeFiles/tdtl.dir/td/tl/tl_file_outputer.cpp.o
[  0%] Building CXX object tdtl/CMakeFiles/tdtl.dir/td/tl/tl_core.cpp.o
[  0%] Building CXX object tdtl/CMakeFiles/tdtl.dir/td/tl/tl_outputer.cpp.o
[  0%] Building CXX object tdutils/CMakeFiles/tdutils.dir/td/utils/port/Clocks.cpp.o
[  0%] Building C object sqlite/CMakeFiles/tdsqlite.dir/sqlite/sqlite3.c.o
[  0%] Building CXX object tdtl/CMakeFiles/tdtl.dir/td/tl/tl_generate.cpp.o
[  1%] Building CXX object tdtl/CMakeFiles/tdtl.dir/td/tl/tl_file_utils.cpp.o
In file included from /Users/vincent/Developer/Resources/Pigeon/Dependencies/TDLib/1.8.18-e79f540-2/tdutils/td/utils/port/Clocks.cpp:11:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/chrono:744:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/calendar.h:13:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/duration.h:14:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__type_traits/common_type.h:14:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__type_traits/decay.h:13:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__type_traits/add_pointer.h:17:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__type_traits/remove_reference.h:13:
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/cstddef:46:5: error: <cstddef> tried including <stddef.h> but didn't find libc++'s <stddef.h> header.           This usually means that your header search paths are not configured properly.           The header search paths should contain the C++ Standard Library headers before           any C Standard Library, and you are probably using compiler flags that make that           not be the case.
#   error <cstddef> tried including <stddef.h> but didn't find libc++'s <stddef.h> header. \
    ^
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/cstddef:59:9: error: no member named 'nullptr_t' in the global namespace
using ::nullptr_t;
      ~~^
In file included from /Users/vincent/Developer/Resources/Pigeon/Dependencies/TDLib/1.8.18-e79f540-2/tdutils/td/utils/port/Clocks.cpp:11:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/chrono:744:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/calendar.h:13:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/duration.h:18:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/limits:827:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/type_traits:456:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__type_traits/is_compound.h:14:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__type_traits/is_fundamental.h:14:
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__type_traits/is_null_pointer.h:24:49: error: use of undeclared identifier 'nullptr_t'; did you mean 'nullptr'?
template <>          struct __is_nullptr_t_impl<nullptr_t> : public true_type {};
                                                ^
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__type_traits/is_null_pointer.h:24:49: error: template argument for template type parameter must be a type
template <>          struct __is_nullptr_t_impl<nullptr_t> : public true_type {};
                                                ^~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__type_traits/is_null_pointer.h:23:17: note: template parameter is declared here
template <class _Tp> struct __is_nullptr_t_impl       : public false_type {};
                ^
In file included from /Users/vincent/Developer/Resources/Pigeon/Dependencies/TDLib/1.8.18-e79f540-2/tdutils/td/utils/port/Clocks.cpp:11:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/chrono:744:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/calendar.h:13:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/duration.h:18:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/limits:827:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/type_traits:538:
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/cstdint:149:5: error: <cstdint> tried including <stdint.h> but didn't find libc++'s <stdint.h> header.           This usually means that your header search paths are not configured properly.           The header search paths should contain the C++ Standard Library headers before           any C Standard Library, and you are probably using compiler flags that make that           not be the case.
#   error <cstdint> tried including <stdint.h> but didn't find libc++'s <stdint.h> header. \
    ^
In file included from /Users/vincent/Developer/Resources/Pigeon/Dependencies/TDLib/1.8.18-e79f540-2/tdutils/td/utils/port/Clocks.cpp:11:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/chrono:744:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/calendar.h:13:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/duration.h:19:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/ratio:83:
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/climits:46:5: error: <climits> tried including <limits.h> but didn't find libc++'s <limits.h> header.           This usually means that your header search paths are not configured properly.           The header search paths should contain the C++ Standard Library headers before           any C Standard Library, and you are probably using compiler flags that make that           not be the case.
#   error <climits> tried including <limits.h> but didn't find libc++'s <limits.h> header. \
    ^
In file included from /Users/vincent/Developer/Resources/Pigeon/Dependencies/TDLib/1.8.18-e79f540-2/tdutils/td/utils/port/Clocks.cpp:11:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/chrono:746:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/convert_to_tm.h:13:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/day.h:15:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/compare:145:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__compare/compare_partial_order_fallback.h:13:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__compare/partial_order.h:14:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__compare/weak_order.h:14:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__compare/strong_order.h:20:
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/cmath:320:5: error: <cmath> tried including <math.h> but didn't find libc++'s <math.h> header.           This usually means that your header search paths are not configured properly.           The header search paths should contain the C++ Standard Library headers before           any C Standard Library, and you are probably using compiler flags that make that           not be the case.
#   error <cmath> tried including <math.h> but didn't find libc++'s <math.h> header. \
    ^
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/cmath:587:17: error: expected unqualified-id
    return std::isnan(__lcpp_x);
                ^
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/math.h:166:5: note: expanded from macro 'isnan'
    ( sizeof(x) == sizeof(float)  ? __inline_isnanf((float)(x))          \
    ^
In file included from /Users/vincent/Developer/Resources/Pigeon/Dependencies/TDLib/1.8.18-e79f540-2/tdutils/td/utils/port/Clocks.cpp:11:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/chrono:746:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/convert_to_tm.h:13:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/day.h:15:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/compare:145:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__compare/compare_partial_order_fallback.h:13:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__compare/partial_order.h:14:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__compare/weak_order.h:14:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__compare/strong_order.h:20:
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/cmath:607:17: error: expected unqualified-id
    return std::isinf(__lcpp_x);
                ^
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/math.h:161:5: note: expanded from macro 'isinf'
    ( sizeof(x) == sizeof(float)  ? __inline_isinff((float)(x))          \
    ^
In file included from /Users/vincent/Developer/Resources/Pigeon/Dependencies/TDLib/1.8.18-e79f540-2/tdutils/td/utils/port/Clocks.cpp:11:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/chrono:746:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/convert_to_tm.h:13:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/day.h:15:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/compare:145:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__compare/compare_partial_order_fallback.h:13:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__compare/partial_order.h:14:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__compare/weak_order.h:14:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__compare/strong_order.h:20:
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/cmath:646:38: error: no member named '__promote' in namespace 'std'
                                std::__promote<_A1, _A2> >::type
                                ~~~~~^
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/cmath:646:48: error: '_A1' does not refer to a value
                                std::__promote<_A1, _A2> >::type
                                               ^
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/cmath:643:17: note: declared here
template <class _A1, class _A2>
                ^
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/cmath:646:58: error: expected unqualified-id
                                std::__promote<_A1, _A2> >::type
                                                         ^
In file included from /Users/vincent/Developer/Resources/Pigeon/Dependencies/TDLib/1.8.18-e79f540-2/tdutils/td/utils/port/Clocks.cpp:11:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/chrono:746:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/convert_to_tm.h:19:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/statically_widen.h:17:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__format/concepts.h:17:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__format/format_parse_context.h:14:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__format/format_error.h:14:
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/cstdlib:90:5: error: <cstdlib> tried including <stdlib.h> but didn't find libc++'s <stdlib.h> header.           This usually means that your header search paths are not configured properly.           The header search paths should contain the C++ Standard Library headers before           any C Standard Library, and you are probably using compiler flags that make that           not be the case.
#   error <cstdlib> tried including <stdlib.h> but didn't find libc++'s <stdlib.h> header. \
    ^
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/cstdlib:132:9: error: target of using declaration conflicts with declaration already in scope
using ::abs _LIBCPP_USING_IF_EXISTS;
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/stdlib.h:132:6: note: target of using declaration
int      abs(int) __pure2;
         ^
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/cmath:353:1: note: conflicting declaration
using ::abs _LIBCPP_USING_IF_EXISTS;
^
In file included from /Users/vincent/Developer/Resources/Pigeon/Dependencies/TDLib/1.8.18-e79f540-2/tdutils/td/utils/port/Clocks.cpp:11:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/chrono:746:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/convert_to_tm.h:19:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/statically_widen.h:17:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__format/concepts.h:17:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__format/format_parse_context.h:14:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__format/format_error.h:15:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/stdexcept:47:
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/exception:199:45: error: field has incomplete type 'exception_ptr'
    _LIBCPP_INLINE_VISIBILITY exception_ptr(nullptr_t) _NOEXCEPT : __ptr_() {}
                                            ^
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/exception:194:24: note: definition of 'std::exception_ptr' is not complete until the closing '}'
class _LIBCPP_TYPE_VIS exception_ptr
                       ^
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/exception:199:5: error: '__abi_tag__' attribute only applies to structs, variables, functions, and namespaces
    _LIBCPP_INLINE_VISIBILITY exception_ptr(nullptr_t) _NOEXCEPT : __ptr_() {}
    ^
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__config:709:37: note: expanded from macro '_LIBCPP_INLINE_VISIBILITY'
#  define _LIBCPP_INLINE_VISIBILITY _LIBCPP_HIDE_FROM_ABI
                                    ^
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__config:688:26: note: expanded from macro '_LIBCPP_HIDE_FROM_ABI'
          __attribute__((__abi_tag__(_LIBCPP_TOSTRING(_LIBCPP_VERSIONED_IDENTIFIER))))
                         ^
In file included from /Users/vincent/Developer/Resources/Pigeon/Dependencies/TDLib/1.8.18-e79f540-2/tdutils/td/utils/port/Clocks.cpp:11:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/chrono:746:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/convert_to_tm.h:19:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/statically_widen.h:17:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__format/concepts.h:17:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__format/format_parse_context.h:14:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__format/format_error.h:15:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/stdexcept:47:
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/exception:199:55: error: expected ';' at end of declaration list
    _LIBCPP_INLINE_VISIBILITY exception_ptr(nullptr_t) _NOEXCEPT : __ptr_() {}
                                                      ^
In file included from /Users/vincent/Developer/Resources/Pigeon/Dependencies/TDLib/1.8.18-e79f540-2/tdutils/td/utils/port/Clocks.cpp:11:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/chrono:746:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/convert_to_tm.h:19:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/statically_widen.h:17:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__format/concepts.h:17:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__format/format_parse_context.h:15:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/string_view:208:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__functional/hash.h:28:
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/cstring:66:5: error: <cstring> tried including <string.h> but didn't find libc++'s <string.h> header.           This usually means that your header search paths are not configured properly.           The header search paths should contain the C++ Standard Library headers before           any C Standard Library, and you are probably using compiler flags that make that           not be the case.
#   error <cstring> tried including <string.h> but didn't find libc++'s <string.h> header. \
    ^
In file included from /Users/vincent/Developer/Resources/Pigeon/Dependencies/TDLib/1.8.18-e79f540-2/tdutils/td/utils/port/Clocks.cpp:11:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/chrono:746:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/convert_to_tm.h:19:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__chrono/statically_widen.h:17:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__format/concepts.h:17:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__format/format_parse_context.h:15:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/string_view:220:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/__string/char_traits.h:24:
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS10.0.sdk/usr/include/c++/v1/cstdio:104:5: error: <cstdio> tried including <stdio.h> but didn't find libc++'s <stdio.h> header.           This usually means that your header search paths are not configured properly.           The header search paths should contain the C++ Standard Library headers before           any C Standard Library, and you are probably using compiler flags that make that           not be the case.
#   error <cstdio> tried including <stdio.h> but didn't find libc++'s <stdio.h> header. \
    ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
/Users/vincent/Developer/Resources/Pigeon/Dependencies/TDLib/1.8.18-e79f540-2/sqlite/sqlite/sqlite3.c:37517:8: warning: "gethostuuid() is disabled." [-W#warnings]
#      warning "gethostuuid() is disabled."
       ^
20 errors generated.
make[2]: *** [tdutils/CMakeFiles/tdutils.dir/td/utils/port/Clocks.cpp.o] Error 1
make[1]: *** [tdutils/CMakeFiles/tdutils.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[  1%] Building CXX object tdtl/CMakeFiles/tdtl.dir/td/tl/tl_string_outputer.cpp.o
[  2%] Building CXX object tdtl/CMakeFiles/tdtl.dir/td/tl/tl_writer.cpp.o
[  2%] Linking CXX static library libtdtl.a
[  2%] Built target tdtl
1 warning generated.
/Users/vincent/Developer/Resources/Pigeon/Dependencies/TDLib/1.8.18-e79f540-2/sqlite/sqlite/sqlite3.c:37517:8: warning: "gethostuuid() is disabled." [-W#warnings]
#      warning "gethostuuid() is disabled."
       ^
1 warning generated.
/Users/vincent/Developer/Resources/Pigeon/Dependencies/TDLib/1.8.18-e79f540-2/sqlite/sqlite/sqlite3.c:37517:8: warning: "gethostuuid() is disabled." [-W#warnings]
#      warning "gethostuuid() is disabled."
       ^
1 warning generated.
[  3%] Linking C static library libtdsqlite.a
[  3%] Built target tdsqlite
make: *** [all] Error 2
```

--- a/CMake/iOS.cmake
+++ b/CMake/iOS.cmake
@@ -189,6 +189,9 @@ if (NOT DEFINED CMAKE_IOS_SDK_ROOT)
 endif (NOT DEFINED CMAKE_IOS_SDK_ROOT)
 set (CMAKE_IOS_SDK_ROOT ${CMAKE_IOS_SDK_ROOT} CACHE PATH "Location of the selected iOS SDK")
 
+# Include C++ Standard Library for Xcode 15 builds.
+include_directories(SYSTEM "${CMAKE_IOS_SDK_ROOT}/usr/include/c++/v1")
+
 # Set the sysroot default to the most recent SDK
 set (CMAKE_OSX_SYSROOT ${CMAKE_IOS_SDK_ROOT} CACHE PATH "Sysroot used for iOS support")
 

--- a/CMake/iOS.cmake
+++ b/CMake/iOS.cmake
@@ -189,9 +189,6 @@ if (NOT DEFINED CMAKE_IOS_SDK_ROOT)
 endif (NOT DEFINED CMAKE_IOS_SDK_ROOT)
 set (CMAKE_IOS_SDK_ROOT ${CMAKE_IOS_SDK_ROOT} CACHE PATH "Location of the selected iOS SDK")
 
-# Include C++ Standard Library for Xcode 15 builds.
-include_directories(SYSTEM "${CMAKE_IOS_SDK_ROOT}/usr/include/c++/v1")
-
 # Set the sysroot default to the most recent SDK
 set (CMAKE_OSX_SYSROOT ${CMAKE_IOS_SDK_ROOT} CACHE PATH "Sysroot used for iOS support")
 
@@ -203,8 +200,14 @@ if (NOT DEFINED IOS_ARCH)
         set (IOS_ARCH "x86_64;arm64")
     elseif (IOS_PLATFORM STREQUAL "WATCHOS")
         set (IOS_ARCH "armv7k;arm64_32;arm64")
+
+        # Include C++ Standard Library for Xcode 15 builds.
+        include_directories(SYSTEM "${CMAKE_IOS_SDK_ROOT}/usr/include/c++/v1")
     elseif (IOS_PLATFORM STREQUAL "WATCHSIMULATOR")
         set (IOS_ARCH "x86_64;arm64")
+
+        # Include C++ Standard Library for Xcode 15 builds.
+        include_directories(SYSTEM "${CMAKE_IOS_SDK_ROOT}/usr/include/c++/v1")
     elseif (IOS_PLATFORM STREQUAL "TVOS")
         set (IOS_ARCH "arm64")
     elseif (IOS_PLATFORM STREQUAL "TVSIMULATOR")

